### DISCUSSION TITLE
Ladekurve

### DISCUSSION THREAD

**Discussion by TheMikeCAD** [hoylabs/OpenDTU-OnBattery#1382] [Ideas]
Hallo zusammen,
hier die Beschreibung einer Idee:

Problem:
Bei schönem Wetter und wenig/keinen Verbrauch ist der Akku sehr schnell voll - nachdem die Anlage versucht möglichst viel Strom ins Netz loszuwerden, schaltet in der nächsten Stufe der Victron ab.
Ergebnis, der Strom bleibt ungenutzt (ob gewünscht oder nicht, sei an der Stelle dahingestellt).

Lösung:
Ladung entlang einer Ladekurve. Überschreitet die Batterieladung zu einer bestimmten Uhrzeit einen voreingestellten Wert, dann geht die Anlage auf Passthrough. So wird vorzeitig mehr Strom ins Netz abgegeben. Dies geschieht so lange, bis der nächste Uhrzeitwert mit Ladestand erreicht wird.
Im Sommer ist das ideal, auch wenn dann nicht immer die max. Ladung erreicht wird. Im Winter ist es auch egal, da die Stop-Schwellen erst gar nicht erreicht werden.
Bei höherem Verbrauch werden die Schwellen nicht erreicht. Einziges Problem tritt auf, wenn Vormittag schönes Wetter herrscht und Nachmittag Regen. Dann kann es sein, dass Vormittags Strom abgeflossen ist, der Nachmittags nicht mehr aufgeholt werden kann.
Einzig eine Verbindung mit sehr guter Wetterprognose würde da Abhilfe schaffen.
(Nach meiner Erfahrung funktioniert das nur sehr eingeschränkt bis gar nicht)

Ich habe diese Lösung mit dem iobroker realisiert, wäre natürlich schöner, wenn es direkt auf dem Board laufen würde.

Was haltet ihr davon?
(Ich hab noch eine relativ alte Firmware laufen, zwecks altem Board. Wenn etwas ähnliches bereits implementiert ist, entschuldige ich mich vorab)

**Comment by hoschiking** [comment]
Es hört sich nicht nach einem Problem an, sondern normalem Verhalten in D bei stark schwankendem Wetter ;-)
Was soll die Lösung sein? Was genau soll OpenDTUoB tun? Das habe noch nicht ganz verstanden

**Comment by AndreasBoehm** [comment]
Wo liegt denn der Unterschied ob ich jetzt über den gesamten Tag verteilt einspeise oder erst ab 12 Uhr?

**Comment by TheMikeCAD** [comment]
Also, ich versuche es nochmal deutlicher:
Sommer - 9:00 Uhr Akkuladestand 50 % (weil wenig Verbrauch in der Nacht)

Das Bestreben der DTU ist nun den Akku mit voller Leistung zu füllen..
Nehmen wir somit an, dass der Akku bei schönem Wetter (entsprechender PV Leistung und Akkugröße) bis Mittag voll ist.
Der Victron wird daraufhin erstmal die Leistung reduzieren und dann völlig abschalten. Somit keine Einspeisung mehr.

Um das zu verhindern, wäre es doch wesentlich schlauer, den Passthrough viel eher zu aktivieren. Dies würde dann bei z.B. 1,5 kW PV Leistung den Akku noch immer mit 700W laden, aber gleichzeitig schon mal 800W ins Netz abgeben. Damit der Akku am Abend trotzdem voll ist, muss der Passthrough-Wert aber ständig angepasst werden.
Dies macht jetzt eben der iobroker bei mir
z.B. 9:00 Uhr - passthrough 50%
12:00 Uhr - passthrough 60%
14:00 Uhr passthrough 70%
16:00 Uhr passthrough 90%

@hoschiking: bitte vollständig lesen
@AndreasBoehm: es macht definitiv einen Unterschied, da Victron ja abschaltet.

**Reply by hoschiking** [reply]
Habe ich, deswegen habe ich aber nichts verstanden, weil ich es sehr genau gelesen habe ;-)
Im zweiten Anlauf ist es verständlich. Das was du beschreibst ist ein typisches Problem von DC gekoppelten Anlagen, u. a. Deswegen habe ich mich dagegen entschieden und lade nur AC gekoppelt, da man nur mit dem Überschuss lädt, d.h. die Einspeisung hat Priorität.

**Reply by AndreasBoehm** [reply]
Ich konnte in meinem System noch nicht feststellen das der Victron Laderegler abschaltet.

Sobald SoC 99% erreicht wurde hat die DTU Passthrough aktiviert und die gesamte Leistung die der Laderegler liefert ging ins Netz. Ende war erst als die Sonne untergegangen ist.

Der Laderegler hat nur herunter geregelt als ich Passthrough deaktiviert hatte und die Spannung der Einstellung im Laderegler entsprochen hat.

Für mich klingt das eher danach als ob deine Passthrough Einstellungen nicht zu den Einstellungen der Spannungen im Laderegler und somit nicht zum Akku passen.

**Reply by AndreasBoehm** [reply]
Das ganze ist hier beschrieben: https://opendtu-onbattery.net/firmware/solar_passthrough/#solar-passthrough-is-off

Ich würde dir empfehlen das ganze nochmal mit der aktuellsten Firmware zu versuchen.

Kannst du uns sonst noch etwas mehr über deine Anlage erzählen?
- max PV Leistung (kWp)
- max Wechselrichter Leistung
- Anzahl und max Leistung der Laderegler
- Akku Größe in kWh

Und gerne auch ein paar Screenshots von deinen Einstellungen, dann können wir besser nachvollziehen wo es Probleme geben könnte.

**Reply by SW-Niko** [reply]
Hallo Leute,
es gibt tatsächlich ein kleines Problem mit Passthrough. Das aber nicht bei jedem auftritt.
Ich habe das hier schon mal genauer beschrieben: #1167  
(Schaut euch im PDF die Tabelle auf der letzten Seite an.)
Kurzgefasst: Wenn der Laderegler sich im Konstant-Spannung-Betrieb befindet, und die (Full) Solar-Passthrough Losses zu klein konfiguriert wurden, dann sinkt die Laderegler Leistung immer weiter ab und es kommt zur Abschaltung. 
Der workaround  ist einfach: Erhöhen der (Full) Solar-Passthrough Losses

**Reply by spcqike** [reply]
> Kannst du uns sonst noch etwas mehr über deine Anlage erzählen?

    max PV Leistung (kWp)
    max Wechselrichter Leistung

das sind natürlich zwei wesentliche Punkte. Wenn ich 2kWp PV Leistung habe, gegen Ladeende aber nur 800W weg bekomme, wird die Batterie auf kurz oder lang voll und der Laderegler geht runter.

> Deswegen habe ich mich dagegen entschieden und lade nur AC gekoppelt,

Das überlege ich auch fürs kommende Jahr. allerdings hat das den Nachteil, dass Overpaneling nicht doch genutzt werden kann. man braucht damit zwangsweise einen überdimensionierten WEchselrichter, oder?

**Reply by hoschiking** [reply]
@spcqike 
Von welchen Wechselrichter sprichst du? 
Ich habe einen Wechselrichter für die Anlage auf dem Dach, Growatt 3kW, dann 2 Balkonkraftwerke 800W mit je 2 Modulen mit jeweils einem HM600, dann noch einen HM300 zur Einspeisung aus dem Speicher. Da kommt einiges zusammen ;-) natürlich muss man es nicht so splitten, bei mir geht es nicht anders, habe nicht viel Platz.
Warum sollte man nicht Overpaneling nicht nutzen können? 2kWp + 800W Wechselrichter ist erlaubt. Es sei denn, du willst mehr als 2kWp montieren.

**Reply by spcqike** [reply]
@hoschiking von allen.
mir schwebt die Erweiterung mittels Victron Multiplus II vor. erstmal einer mit 5kW, später dann 3.

was ich meine: wenn man nun 2kWp auf dem Dach hat und einen 800VA Wechselrichter, dahinter (bzw. eher daneben) einen AC gekoppelten Speicher wie den Multiplus oder sonst was packt, "drosselt" man seinen "Überschuss" auf maximal 800W. eben das, was der 800VA Wechselrichter hergibt.
ein DC MPPT Laderegler wie die Smartsolar hingegen kann die 2kWp voll in die Batterie laden, nutzt also das volle Potential der Module.
Will man also keine künstliche Drossel einbauen, müssen die Wechselrichter größer sein, als die Module. Auch muss der AC Speicher/Laderegler größer sein, als die anderen Wechselrichter. 

also klar, man kann den Hoymiles überbelegen. Im Herbst und Winter super. Aber im Sommer verliert man halt, verglichen mit einem DC Lader, viel Energie, die man dann nicht mit in die Nacht nehmen kann.


Vielleicht verkompliziere ich das alles bereits wieder zu sehr :D ich habe halt aktuell 8x435Wp und einen HMS-2000. im Sommer bleibt der eben bei 2000W stehen. die restlichen bis zu 1.4kWp bleiben ungenutzt. ein MPPT 150/60 könnte die 3.4kWp voll nutzen, würde dann aber früher oder später abregeln, wie hier, und ich müsste alles in der kalten Garage aufbauen. Daher (und auch weil kommende dezentrale Erweiterungen mit zusätzlichen Wechselrichtern einfacher wird) überlege ich eben auch die AC gekoppelte Richtung einzuschlagen. den HMS-2000 durch 2xHMS-1600 zu ersetzen (oder nur einen HMS-1600 dazu zu holen) und an anderer Stelle im Haus den Multiplus samt Speicher zu platzieren.

**Comment by hoschiking** [comment]
@spcqike 
jetzt verstehe ich es. Das kann man jedoch elegant umgehen, so mache ich es ja auch.
Die Wechselrichterleistung darf ja höher sein, man darf nur nicht mehr als 800W einspeisen, also muß man drosseln.
Du kannst ja die Drosselung dynamisch gestalten, wenn du lädst, regelst du die Wechselrichter hoch, sobald der Speicher voll ist,
kannst du wieder runter regeln. Die Steuerung  ist ja dank OpenDTUoB ziemlich einfach ;-)
Bei mir läuft dafür ein kleines Script im ioBroker.
Mittlwerweile kann man die Begrenzung direkt per MQTT "Topic/powerlimiter/cmd/upper power limit" vorgeben.
Wenn ich die Leistung nicht (mehr) benötige, setzte ich den Wert für beide Wechselrichter auf 800W (300W und 500W).
Der größte Nachteil DC-gekoppelt ist ja, daß die gesamte Leistung nicht beliebig verwertet werden kann, weil fragmentiert, siehe auch diese Diskussion und man räumlich eingeschränkt ist. AC-gekoppelt hat man die räumliche Einschränkungen nicht, bei mir gibt es 3 Anlagen, eine auf dem Hausdach, eine auf dem Carport und eine im Garten als Überdachung des Holzunterstandes. Gesamte PV-Leistung landet im Netz und die kann ich per ioBroker so vewrwerten, wie gerade benötigt. Beispiel: wenn die Waschmaschine gerade heizt (2000W) und die PV-Leistung nicht zum gleichzeitigen Laden reicht, unterbreche ich die Ladung einfach. Alle benötigten Daten liegen im MQTT-Broker sowieso vor. Die einzigige Hürde ist die dynamische Regelung der Ladeleistung je nach PV-Überschuß. Ich als Softwarelegasteniker tue mich da schwer und komme nicht wirklich weiter. Bisher habe ich einfach eine Ladereglerkaskade und schalte die nach Bedarf ein und aus. Und das Problem im Sommer ist rein theoretischer Natur. Obwohl meine Ladeleistung durch den Aufbau begrenz ist (820W auf der DC Seite), war der Speicher (24V / 6kWh) praktisch vom Anfang März bis Ende Oktober niemals leer. Pünklich zum 1 November war das erste Mal Ebbe, aber nur weil nicht genug PV-Leistung vorhanden war, seit dem ist es fast täglich nur noch dunkelgrau. Wenn es einen sonnigen Tag gab, war der Speicher an diesem einen Tag wieder randvoll. Auf der anderen Seite spricht auch nichts gegen einen Mischbetrieb, einige Module DC-gekoppelt + Überschußladen AC-gekoppelt. Solange du Victron-Laderegler nimmst und diese synchron Laden, ist der Mehraufwand zur Regelung gleich 0. Ich habe 1 Jahr lang immer mehr Erfahrungen gesammelt und die Anlage entsprechend modifiziert. Für Erfahrungsaustausch bin ich immer zu haben. Schön wäre es, wenn mich auch jemand unterstützen könnte den Ladestrom vom MPPT per VE.Direct  dynamisch regeln zu können ;-)
